packages/aws-cdk-lib/aws-events-targets/lib/redshift-query.ts

```diff
@@ -0,0 +1,134 @@
+   * If the full secret ARN is not specified, this will instead use the secret name.
+   *
+   * @default - No secret is specified
+   */
+  readonly secret?: secretsmanager.ISecret;
+
+  /**
+   * The SQL query to run. This will use the `executeStatement` API.
+   *
+   * @default - No SQL query is specified
+   */
+  readonly sql?: string;
+
+  /**
+   * The SQL queries to run. All statements are executed as a single transaction. They each run serially, as appeared in the array; the next sql statement will not run until the previous statement completes.
+   * If any statement fails, the entire transaction is rolled back. This will use the `batchExecuteStatement` API.
+   *
+   * @default - No SQL queries are specified
+   */
+  readonly batchSQL?: string[];
+
+  /**
+   * The name of the SQL statement. You can name the SQL statement for identitfication purposes. It is recommended to append a `QS2-` prefix to the statement name, to allow Amazon Redshift to identify the Event Bridge rule, and present it in the Amazon Redshift console.
+   *
+   * @default - No statement name is specified
+   */
+  readonly statementName?: string;
+
+  /**
+   * Should an event be sent back to Event Bridge when the SQL statement is executed.
+   *
+   * @default - false
+   */
+  readonly withEvent?: boolean;
+
+  /**
+   * The queue to be used as dead letter queue.
+   *
+   * @default - No dead letter queue is specified
+   */
+  readonly deadLetterQueue?: sqs.IQueue;
+
+  /**
+   * The IAM role to be used to execute the SQL statement.
+   *
+   * @default - a new role will be created.
+   */
+  readonly role?: iam.IRole;
+
+  /**
+   * The input to the state machine execution
+   *
+   * @default the entire EventBridge event
+   */
+  readonly input?: events.RuleTargetInput;
+}
+
+/**
+ * Schedule an Amazon Redshift Query to be run, using the Redshift Data API.
+ * It is recommended to append a `QS2-` prefix to both `statementName` and `ruleName`, to allow Amazon Redshift to identify the Event Bridge rule, and present it in the Amazon Redshift console.
+ */
+export class RedshiftQuery implements events.IRuleTarget {
+  constructor(
+    private readonly clusterArn: string,
+    private readonly props: RedshiftQueryProps = {}) {
+  }
+
+  bind(rule: events.IRule, _id?: string): events.RuleTargetConfig {
+    const role = this.props.role ?? singletonEventRole(rule);
+    if (this.props.sql) {
+      role.addToPrincipalPolicy(this.putEventStatement());
+    }
+    if (this.props.batchSQL) {
+      role.addToPrincipalPolicy(this.putBatchEventStatement());
+    }
+
+    return {
+      ...bindBaseTargetConfig(this.props),
+      arn: this.clusterArn,
+      role,
+      input: this.props.input,
+      redshiftDataParameters: {
+        database: this.props.database ?? 'dev',
+        dbUser: this.props.dbUser,
+        secretManagerArn: this.props.secret?.secretFullArn ?? this.props.secret?.secretName,
```

Comment:
Could you help me understand this line? It's not clear to me why `secretFullArn` is not always there and why the default should be `secretName`.

Reply:
Hi @Rizxcviii, just pinging about these two questions.

Reply:
> Also can we add a unit test where secretFullArn is not specified and secret.secretName is used?

Hey @paulhcsun, sorry for not seeing this earlier. I can add the unit test.
> Since secret is optional, is there a chance that this would not be set at all?

Yes, according to the docs
https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_RedshiftDataParameters.html

> The name or ARN of the secret that enables access to the database. Required when authenticating using AWS Secrets Manager.

So from my understanding, there will be times when a developer does not need to specify a set of AWS Secrets Manager credentials.

Reply:
@paulhcsun, I've added them in

Reply:
Gotcha, thanks for following up on that! The tests look good, I think the only thing left is the `sql` property but I'd like to get the input of the Redshift service team on that first. Once I hear back and we decide on an implementation here this PR should be good to merge :ok_hand: